File: gymgenie-mobile/composeApp/src/androidMain/kotlin/com/asc/gymgenie/feature/workouts/components/WorkoutCardSmall.kt
package com.asc.gymgenie.feature.workouts.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.defaultMinSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.asc.gymgenie.ui.theme.AccentOrange
import com.asc.gymgenie.ui.theme.DeepInk
import com.asc.gymgenie.ui.theme.MutedText
import com.asc.gymgenie.ui.theme.SoftCard
import com.asc.gymgenie.workout.WorkoutPlanShortResponse

@Composable
fun WorkoutCardSmall(
    plan: WorkoutPlanShortResponse,
    onStart: () -> Unit = {},
) {
    val isAI = plan.createdBy.equals("AI", ignoreCase = true)
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .defaultMinSize(minHeight = 160.dp)
            .shadow(2.dp, RoundedCornerShape(16.dp))
            .clip(RoundedCornerShape(16.dp))
            .background(Color.White)
            .then(
                if (plan.isActive) Modifier.border(
                    width = 1.5.dp,
                    color = AccentOrange,
                    shape = RoundedCornerShape(16.dp),
                ) else Modifier
            )
            .padding(14.dp),
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.Top,
        ) {
            Text(
                text = plan.name,
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold,
                color = DeepInk,
                modifier = Modifier.weight(1f),
                maxLines = 2,
                overflow = TextOverflow.Ellipsis,
            )
            if (isAI) {
                Spacer(modifier = Modifier.width(6.dp))
                Text(
                    text = "+ AI",
                    fontSize = 9.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White,
                    modifier = Modifier
                        .clip(RoundedCornerShape(50))
                        .background(AccentOrange)
                        .padding(horizontal = 8.dp, vertical = 3.dp),
                )
            }
        }

        Spacer(modifier = Modifier.height(8.dp))

        Text(
            text = "${plan.daysCount} дней/нед.",
            fontSize = 11.sp,
            color = MutedText,
            modifier = Modifier
                .clip(RoundedCornerShape(50))
                .background(SoftCard)
                .padding(horizontal = 10.dp, vertical = 4.dp),
        )

        if (plan.isActive) {
            Spacer(modifier = Modifier.height(8.dp))
            Row(verticalAlignment = Alignment.CenterVertically) {
                Box(
                    modifier = Modifier
                        .size(6.dp)
                        .clip(CircleShape)
                        .background(AccentOrange),
                )
                Spacer(modifier = Modifier.width(5.dp))
                Text(
                    text = "АКТИВНЫЙ",
                    fontSize = 9.sp,
                    fontWeight = FontWeight.Bold,
                    color = AccentOrange,
                )
            }
        }

        Spacer(modifier = Modifier.weight(1f))
        Spacer(modifier = Modifier.height(12.dp))

        Button(
            onClick = onStart,
            modifier = Modifier.fillMaxWidth(),
            colors = ButtonDefaults.buttonColors(
                containerColor = AccentOrange,
                contentColor = Color.White,
            ),
            shape = RoundedCornerShape(10.dp),
            contentPadding = PaddingValues(vertical = 8.dp),
        ) {
            Text(
                text = "Начать",
                fontSize = 12.sp,
                fontWeight = FontWeight.SemiBold,
            )
        }
    }
}

File: gymgenie-mobile/iosApp/iosApp/Home/WorkoutCardSmall.swift
import SwiftUI
import Shared

struct WorkoutCardSmall: View {
    let plan: WorkoutPlanShortResponse
    var onStart: () -> Void = {}

    private let orange = Color(red: 0.941, green: 0.439, blue: 0.188)
    private let deepInk = Color(red: 0.161, green: 0.141, blue: 0.125)
    private let softCard = Color(red: 0.953, green: 0.949, blue: 0.937)
    private let mutedText = Color(red: 0.463, green: 0.447, blue: 0.416)

    private var isAI: Bool {
        plan.createdBy.uppercased() == "AI"
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack(alignment: .top) {
                Text(plan.name)
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(deepInk)
                    .lineLimit(2)
                    .frame(maxWidth: .infinity, alignment: .leading)

                if isAI {
                    Text("+ AI")
                        .font(.system(size: 9, weight: .bold))
                        .foregroundColor(.white)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 3)
                        .background(Capsule().fill(orange))
                }
            }

            Text("\(plan.daysCount) дней/нед.")
                .font(.system(size: 11))
                .foregroundColor(mutedText)
                .padding(.horizontal, 10)
                .padding(.vertical, 4)
                .background(Capsule().fill(softCard))

            if plan.isActive {
                HStack(spacing: 5) {
                    Circle().fill(orange).frame(width: 6, height: 6)
                    Text("АКТИВНЫЙ")
                        .font(.system(size: 9, weight: .bold))
                        .foregroundColor(orange)
                }
            }

            Spacer(minLength: 4)

            Button(action: onStart) {
                Text("Начать")
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundColor(.white)
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 8)
                    .background(RoundedRectangle(cornerRadius: 10).fill(orange))
            }
            .buttonStyle(.plain)
        }
        .padding(14)
        .frame(maxWidth: .infinity, minHeight: 150, alignment: .leading)
        .background(RoundedRectangle(cornerRadius: 16).fill(.white))
        .overlay(
            RoundedRectangle(cornerRadius: 16)
                .stroke(plan.isActive ? orange : Color.clear, lineWidth: 1.5)
        )
        .shadow(color: Color.black.opacity(0.04), radius: 4, y: 2)
    }
}
